### ISSUE TITLE
Use a different example to explain Appendable Things

### ISSUE THREAD

**Issue by DaniRey** [fommil/fpmortals#4]
In the chapter "## Appendable Things" I struggled with the TradeTemplate example for two reasons. It doesn't make sense to me, to append TradeTemplates. It doesn't feel natural that one should append TradeTemplates of different Currencies.
Also, I didn't know what OTC means (after looking it up, it is probably "over the counter").

So I suggest you either choose a different example, or explain the TradeTemplate and its purpose in more detail.

Thinking about it I came up with an example based on a Bid. Assuming that the Bids would be appended in correct order, this would make sense to me.

{lang="text"}
~~~~~~~~
  type Name = String
  sealed abstract class BidderKind
  case object Remote extends BidderKind
  case object Local extends BidderKind
  
  final case class Bid (
    bids: List[Int],
    highest: Option[Name],
    kind: Option[BidderKind]
  )
~~~~~~~~

**Comment by fommil** [comment]
I've added more text around the example

```
As a realistic example for =Monoid=, consider a trading system that has a large
database of reusable trade templates. Populating the default values for a new
trade involves selecting and combining multiple templates, with a "last rule
wins" merge policy if two templates provide a value for the same field. The
"selecting" work is already done for us by another system, it is our job to
combine the templates in order.

We'll create a simple template schema to demonstrate the principle,
but keep in mind that a realistic system would have a more complicated
ADT.
```

But this is the fundamental problem about using "real world examples". Actually what people want is "examples that are in my domain" and even if they are **slightly** outside that domain (like we have here) it becomes more of a blocker than a helper. In fact, the more somebody knows about the domain, the less convincing the example is, because the real world is messy and examples like this should explain the principle not the real world application.

Basically the whole "trade" thing is irrelevant (although this is actually real world example). We just have some `case class` that we need to add together.

**Comment by DaniRey** [comment]
You are right. I think the additional text helps a lot to grasp the context of the example.

Maybe you could add a comment for OTC in the case class.

`
  final case class TradeTemplate(
    payments: List[java.time.LocalDate],
    ccy: Option[Currency],
    // over the counter
    otc: Option[Boolean]
  )`

**Comment by fommil** [comment]
that's going too far... the reader doesn't need to understand that. It might even be better if I make something up instead of using `otc`.

**Comment by DaniRey** [comment]
Good point. What about `international` instead of `otc`?

I guess trades could be national or international. No domain specific knowledge required to see that a boolean is a good way to represent this.

**Comment by fommil** [comment]
until somebody comes along and asks why `international` is there, because they'd expect to see `otc` :laughing:
